machine/delta/delta_wb2448/kernel/igb-bcm50210s.patch

```diff
@@ -0,0 +1,193 @@
+diff --git a/drivers/net/ethernet/intel/igb/e1000_82575.c b/drivers/net/ethernet/intel/igb/e1000_82575.c
+index 0f69ef8..6311d8b 100644
+--- a/drivers/net/ethernet/intel/igb/e1000_82575.c
++++ b/drivers/net/ethernet/intel/igb/e1000_82575.c
+@@ -166,6 +166,7 @@ static s32 igb_init_phy_params_82575(struct e1000_hw *hw)
+ {
+ 	struct e1000_phy_info *phy = &hw->phy;
+ 	s32 ret_val = 0;
++    u32 u32Data = 0;
+ 	u32 ctrl_ext;
+ 
+ 	if (hw->phy.media_type != e1000_media_type_copper) {
+@@ -186,6 +187,12 @@ static s32 igb_init_phy_params_82575(struct e1000_hw *hw)
+ 		ctrl_ext &= ~E1000_CTRL_I2C_ENA;
+ 	}
+ 
++    /* change to mdio mode */
++    u32Data = rd32(0xe04);
++    u32Data |= 0x80000000;
++    wr32(0xe04, u32Data);
++
++
+ 	wr32(E1000_CTRL_EXT, ctrl_ext);
+ 	igb_reset_mdicnfg_82580(hw);
+ 
+@@ -286,6 +293,8 @@ static s32 igb_init_phy_params_82575(struct e1000_hw *hw)
+ 		phy->ops.set_d3_lplu_state = igb_set_d3_lplu_state_82580;
+ 		phy->ops.force_speed_duplex = igb_phy_force_speed_duplex_m88;
+ 		break;
++    case BCMB50210S_PHY_ID:
++        break;
+ 	default:
+ 		ret_val = -E1000_ERR_PHY;
+ 		goto out;
+diff --git a/drivers/net/ethernet/intel/igb/e1000_defines.h b/drivers/net/ethernet/intel/igb/e1000_defines.h
+index 217f813..f13777a 100644
+--- a/drivers/net/ethernet/intel/igb/e1000_defines.h
++++ b/drivers/net/ethernet/intel/igb/e1000_defines.h
+@@ -860,6 +860,7 @@
+ #define M88_VENDOR           0x0141
+ #define I210_I_PHY_ID        0x01410C00
+ #define M88E1543_E_PHY_ID    0x01410EA0
++#define BCMB50210S_PHY_ID    0x600d8590
+ 
+ /* M88E1000 Specific Registers */
+ #define M88E1000_PHY_SPEC_CTRL     0x10  /* PHY Specific Control Register */
+diff --git a/drivers/net/ethernet/intel/igb/igb_main.c b/drivers/net/ethernet/intel/igb/igb_main.c
+index 7dfbcde..7bd6c9d 100644
+--- a/drivers/net/ethernet/intel/igb/igb_main.c
++++ b/drivers/net/ethernet/intel/igb/igb_main.c
+@@ -68,6 +68,10 @@ static const char igb_driver_string[] =
+ static const char igb_copyright[] =
+ 				"Copyright (c) 2007-2014 Intel Corporation.";
+ 
++#if 1 /* add BCM B50210S phy combo mode initialize */
```

Comment:
Get rid of all these `#if 1 / #endif`.

If the code being bracketed needs to be there then put it in the patch -- no need to bracket it with `#if 1 / #endif`.  That is just cruft and does not belong in production code.